Makefile

```diff
@@ -164,7 +163,7 @@ publish:
 	)
 
 check_codegen:
-	command $(CODEGEN) > /dev/null 2>&1 || { echo $CODEGEN_INSTALL_MSG; false; }
+	test -s ${SWAGGER_JAR} || wget  ${SWAGGER_DOWNLOAD_URL} -O ${SWAGGER_JAR}
```

Comment:
The .gitignore entry looks good.  I was just thinking of something like:

```
	${SWAGGER_JAR}:
		wget  ${SWAGGER_DOWNLOAD_URL} -O $@
```

Reply:
Thanks for the clarification! Let me know how it looks now. I still think we want the `test -s $@ ||` to avoid downloading the jar repeatedly.

Reply:
`make` shouldn't rebuild the rule if the file exists.

Reply:
I think I have it sorted now.